### PR TITLE
feat/new DQS report download

### PR DESCRIPTION
**Pull request by pritish-ranjan01**

### CODE REVIEW

File: transit_odp/common/utils/s3_bucket_connection.py [hunk 1]
```diff
@@ -40,6 +42,44 @@ def get_s3_bodds_bucket_storage():
         raise
 
 
+def get_dqs_report_from_s3(report_filename):
```
Comment (NitishAroraKPMG) [inline transit_odp/common/utils/s3_bucket_connection.py:45]: documentation

File: transit_odp/common/utils/s3_bucket_connection.py [hunk 2]
```diff
@@ -40,6 +42,44 @@ def get_s3_bodds_bucket_storage():
...
+    bucket_name = getattr(settings, "S3_BUCKET_DQS_CSV_REPORT", None)
+    if not bucket_name:
+        logger.error("Bucket name is not configured in settings.")
```
Comment (NitishAroraKPMG) [inline transit_odp/common/utils/s3_bucket_connection.py:48]: env. value - S3_BUCKET_DQS_CSV_REPORT is missing

File: transit_odp/common/utils/s3_bucket_connection.py [hunk 3]
```diff
@@ -40,6 +42,44 @@ def get_s3_bodds_bucket_storage():
...
+        error_code = e.response["Error"]["Code"]
+        if error_code == "403":
+            logger.error(
+                f"Permission denied when accessing the S3 bucket: {bucket_name}"
+            )
+            return HttpResponse(
+                "Permission denied when accessing the S3 bucket", status=403
+            )
+        else:
+            logger.error(f"Error connecting to S3 bucket {bucket_name}: {str(e)}")
```
Comment (NitishAroraKPMG) [inline transit_odp/common/utils/s3_bucket_connection.py:76]: add error code in the logging also

**Review by amitkpmguk — APPROVED** [review]
Looks good to me 👍

**Review by amitkpmguk — APPROVED** [review]
Looks good to me 👍

**Pull request merged**